### PR TITLE
Testing updates

### PR DESCRIPTION
**Pull request by hategan**
This PR contains a bunch of more-or-less minor updates to testing:
- always check out a current ci-runner from main
- move non-venv non-conda dep installs from --user to .packages inside the testing setup dir
- install/upgrade dependencies on every branch that is tested
- add a requirements-tests.txt, which installs only dependencies needed for testing

### CODE REVIEW

File: psij-ci-setup [hunk 1]
```diff
@@ -51,11 +58,19 @@ done
...
+if [ "$MODE" != "plain" ]; then
+    OUT=`$PIP install -r requirements-tests.txt 2>&1`
 else
-    $PIP install --user -r requirements-dev.txt
+    OUT=`$PIP install --target .packages --upgrade -r requirements-tests.txt 2>&1`
+fi
+
+if [ "$?" != "0" ]; then
+    echo "FAILED"
+    echo $OUT
```
Comment (andre-merzky) [inline psij-ci-setup:71]: Should we exit here?

**Review by andre-merzky — APPROVED** [review]
One minor comment, otherwise LGTM - thanks!

File: psij-ci-setup (hunk 1 above)
Comment (hategan) [reply psij-ci-setup:69]: We should.

**Review by jameshcorbett — APPROVED** [review]
LGTM

**Pull request merged**